api/client/run.go

```diff
@@ -260,6 +261,16 @@ func (cli *DockerCli) CmdRun(args ...string) error {
 
 	// Attached mode
 	if *flAutoRemove {
+		// Warn user if they detached us
+		js, err := cli.client.ContainerInspect(createResponse.ID)
+		if err != nil {
+			return runStartContainerErr(err)
```

Comment:
My take is that if we fail here, the next api call is mostly certain to fail too.